grails-app/controllers/au/org/emii/netcdfsubset/SubsetController.groovy

```diff
@@ -21,13 +23,29 @@ class SubsetController {
         response.setContentType("application/octet-stream")
         response.setHeader("Content-disposition", "filename=${filenameToServe(subsetRequest)}")
 
-        try {
-            subsetService.subset(subsetRequest.typeName, subsetRequest.CQL_FILTER, response.outputStream)
+        if (isThreadAvailable()) {
```

Comment:
Is there a race condition between the testing of the mutex in isThreadAvailable(), and acquiring it in getThreadLock()? Eg. two or more threads could pass through isThreadAvailable() before getting to the next statement.